llvm/lib/ProfileData/InstrProfWriter.cpp

```diff
@@ -604,6 +608,35 @@ Error InstrProfWriter::writeImpl(ProfOStream &OS) {
       OS.writeByte(0);
   }
 
+  // if version >= the version with vtable profile metadata.
+  uint64_t VTableNamesSectionStart = 0;
+  if (IndexedInstrProf::ProfVersion::CurrentVersion >= 12) {
```

Comment:
Is the check needed? Instead add an assertion on the version.

Reply:
The check is verbose. I removed it.